### PR TITLE
Fixed makeUnPushable/unDroppable bug

### PR DESCRIPTION
**Pull request by jSwanson99**
… and added a loading screen/overlay while waiting for view to mount

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "aframe-react": "^4.4.0",
     "bootstrap": "^4.0.0",
     "cannon": "^0.6.2",
-    "create-react-app": "^1.5.2",
+    "create-react-app": "^3.0.1",
```
Comment (jasondkiesling) [inline package.json]: Was there a particular reason for updating this?

File: src/components/Header.js [hunk 2]
```diff
@@ -30,6 +30,8 @@ const exitBtnStyle = {
   right: 0,
 };
 
+let shouldRender = false;
```
Comment (jasondkiesling) [inline src/components/Header.js]: This should be part of the component state. Then whenever you need to change it you should use this.setState({}). Having anything outside of the class that isn't a const can cause issues.

File: src/components/Header.js [hunk 3]
```diff
@@ -180,6 +182,10 @@ class Header extends Component {
     if (this.state.lastMsgTime !== this.props.message.time && this.props.message.text !== "") {
       this.setState({ snackOpen: true, lastMsgTime: this.props.message.time });
     }
+    if(shouldRender) {
```
Comment (jasondkiesling) [inline src/components/Header.js]: This doesn't look like it is really the way that we should be doing this. Something is weird.

File: src/components/View.js [hunk 4]
```diff
@@ -12,15 +12,24 @@ import * as THREE from 'three';
  * system utilizes the entity compoent system(ECS) to build objects in the scene from different
  * components.
  */
+
+ let isLoading = true;
```
Comment (jasondkiesling) [inline src/components/View.js]: This also should be part of the component state.

File: src/components/View.js [hunk 5]
```diff
@@ -169,7 +178,7 @@ class View extends Component {
   }
 
   render = () => {
-    return (
+    const aScene = (
```
Comment (jasondkiesling) [inline src/components/View.js]: Is this actually a const?

File: src/components/View.js [hunk 6]
```diff
@@ -200,6 +208,21 @@ class View extends Component {
...
       </a-scene>
     );
+  
+    const spinner = 
+      <span className='spinner'>
+        <div className='cube1'></div>
+        <div className='cube2'></div>
+      </span>
+
+    let ret = null;
```
Comment (jasondkiesling) [inline src/components/View.js]: This can be reduced to `return isLoading ? spinner : aScene;`.

File: src/components/View.js (hunk 5 above)
Comment (jasondkiesling) [inline src/components/View.js]: This variable name is also unclear. Let's make it more clear.

File: src/components/layouts/Ide.js [hunk 7]
```diff
@@ -31,9 +31,9 @@ export const Ide = ({ editor, editorActions, user, authActions, scene, sceneActi
             {
                 scene.settings.viewOnly
                     ?
-                    <div id="scene" className="col-12" >
-                        <View objects={editor.objects} sceneConfig={scene} assets={editor.assets} />
-                    </div>
+                        <div id="scene" className="col-12" >
```
Comment (jasondkiesling) [inline src/components/layouts/Ide.js]: Was this run through the formatter?

File: src/myr/Myr.js [hunk 8]
```diff
@@ -289,17 +289,39 @@ class Myr {
   makeDroppable = (outerElId, mass = 2) => {
     let el = this.getEl(outerElId);
     let dynamicBody = `shape: auto; mass: ${mass}; angularDamping: 0.5; linearDamping: 0.5;`;
-    el["dynamic-body"] = dynamicBody;
+    el["dynamic-body"] = dynamicBody;
```
Comment (jasondkiesling) [inline src/myr/Myr.js]: Please remove unneeded whitespace.

File: src/myr/Myr.js [hunk 9]
```diff
@@ -289,17 +289,39 @@ class Myr {
...
     let el = this.getEl(outerElId);
-    el["dynamic-body"] = "";
+    //Only makes an item undroppable if it is droppable but is not pushable
+    if(el["dynamic-body"] && (!el["force-pushable"] || el["force-pushable"] === "false")) {
+      el["dynamic-body"] = null;
+    }
     return outerElId;
   }
 
+  // Allows the entity to be pushed
```
Comment (jasondkiesling) [inline src/myr/Myr.js]: Was this run through the formatter? There is an option to auto-format on save that is best if it is turned on.

File: yarn.lock [hunk 10]
```diff
@@ -2314,7 +2314,15 @@ chalk@2.4.1:
     escape-string-regexp "^1.0.5"
     supports-color "^5.3.0"
 
-chalk@^1.1.1, chalk@^1.1.3:
```
Comment (jasondkiesling) [inline yarn.lock]: What's the reason for updating all of these?

File: package.json (hunk 1 above)
Comment (jSwanson99) [reply package.json]: I was having problems with create react app after installing eslint, so I had no remove it then install create-react-app again. I can try and go to an older version if needed

File: src/components/View.js (hunk 5 above)
Comment (jSwanson99) [reply src/components/View.js]: I'm not sure what you mean

File: yarn.lock (hunk 10 above)
Comment (jSwanson99) [reply yarn.lock]: They might have been updated when I reinstalled create-react-app, I didn't intentionally update them.

**Comment by jSwanson99** [comment]
I removed the loading screen because I couldn't get it to work through the components state, so I want to take more time to figure out a fix for that. Also any white space that was added/remove in this commit was due to the auto formatter

**Comment by jasondkiesling** [comment]
The way that render is working now isn't really the way we should be doing it. More investigation is needed. If you want to remove that part for now we can push the makeUnDrop/Pushable part.

**Comment by jasondkiesling** [comment]
You also have some merge conflicts that need to be resolved.

**Pull request merged**